src/lib.rs

```diff
@@ -1498,7 +1498,7 @@ fn encode_block(seq: &Sequence, fi: &FrameInvariants, fs: &mut FrameState, cw: &
 
     let tx_type = if tx_set > TxSet::TX_SET_DCTONLY && fi.config.speed <= 3 {
         // FIXME: there is one redundant transform type decision per encoded block
-        rdo_tx_type_decision(fi, fs, cw, luma_mode, bsize, bo, tx_size, tx_set)
+        rdo_tx_type_decision(fi, fs, cw, luma_mode, bsize, bo, tx_size, tx_set, skip)
```

Comment:
Because skip==true (this flag is defined for a partition) means coefficient are not coded, so the transform is NOT applied, the skip flag should be false when rdo_tx_type_decision() is called.
I think you might be confused with two different kinds of skip, one for tx block (which is coded a part of coefficient coding) and another for partition, which is coded once per each partition.